File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> table;
        vector<int> res(2);
        int t, count = 0;
        
        for(auto itr = nums.begin(); itr != nums.end(); itr++){
            
            t = target - *itr;
            
            if(table.find(*itr) == table.end())
                table[t] = count;
            else{
                res[0] = table[*itr];
                res[1] = count;
                return res;
            }
            
            count++;
        }
        
        return res;  
    }
};

File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    
    table := make(map[int]int)
    res := []int{}
    
    for idx, val := range nums {
        
        t := target - val
                
        if _, ok := table[t]; ok {
            
            res = append(res, table[t], idx)
        }
        
        table[val] = idx
    }
    
    return res
}